cyclonedx/output/json.py

```diff
@@ -97,44 +97,43 @@ def generate(self, force_regeneration: bool = False) -> None:
         self.generated = True
 
     def _specialise_output_for_schema_version(self, bom_json: Dict[Any, Any]) -> str:
-        if not self.bom_supports_metadata():
-            if 'metadata' in bom_json.keys():
+        if 'metadata' in bom_json.keys():
+            if not self.bom_supports_metadata():
                 del bom_json['metadata']
+            else:
+                if 'tools' in bom_json['metadata'].keys():
+                    if not self.bom_metadata_supports_tools():
+                        del bom_json['metadata']['tools']
+                    else:
+                        if not self.bom_metadata_supports_tools_external_references():
+                            for _tool in bom_json['metadata']['tools']:
+                                if 'externalReferences' in _tool.keys():
+                                    del _tool['externalReferences']
+                                del _tool
+                if 'licenses' in bom_json['metadata'].keys() and not self.bom_metadata_supports_licenses():
+                    del bom_json['metadata']['licenses']
+                if 'properties' in bom_json['metadata'].keys() and not self.bom_metadata_supports_properties():
+                    del bom_json['metadata']['properties']
+
+                if self.get_bom().metadata.component:
+                    bom_json['metadata'] = self._recurse_specialise_component(bom_json['metadata'], 'component')
+
+        bom_json = self._recurse_specialise_component(bom_json)
 
-        if not self.bom_metadata_supports_tools():
-            del bom_json['metadata']['tools']
```

Comment:
this was a delete without checking if `bom_json['metadata']` existed in the first place ...